subprojects/docs/src/docs/userguide/build-cache/build_cache.adoc

```diff
@@ -328,6 +328,24 @@ include::sample[dir="snippets/buildCache/http-build-cache/groovy",files="setting
 include::sample[dir="snippets/buildCache/http-build-cache/kotlin",files="settings.gradle.kts[tags=allow-untrusted-server]"]
 ====
 
+[[sec:build_cache_expect_continue]]
+The HTTP build cache client allows opt-in use of https://www.w3.org/Protocols/rfc2616/rfc2616-sec8.html#sec8.2.3[HTTP Expect-Continue].
+This causes PUT requests to happen in two parts: first a check whether a body would be accepted, then transmission of the body if the server indicates it will accept it.
+This is particularly suitable for cache servers that routinely redirect or reject PUT requests,
+as it avoids transmitting the cache entry just to have it rejected (e.g. the cache entry is larger than the cache will allow).
+This additional check incurs extra latency when the server accepts the request, but reduces latency when the request is rejected.
+
+Not all HTTP servers and proxies reliably implement Expect-Continue.
+Be sure to check that your cache server does support it before enabling.
+
+To opt-in, set link:{groovyDslPath}/org.gradle.caching.http.HttpBuildCache.html#org.gradle.caching.http.HttpBuildCache:useExpectContinue[HttpBuildCache.isUseExpectContinue()] to `true`.
```

Comment:
Should this link to the getter or the setter?

Reply:
Getter, AFAIK.

Reply:
The doc talks about setting the value to `true`, that's why it is a bit strange for me to link to the getter. But if you say it's fine, I'm fine too. :)